src/Avalonia.Controls/AppBuilderBase.cs

```diff
@@ -21,33 +22,48 @@ public abstract class AppBuilderBase<TAppBuilder> where TAppBuilder : AppBuilder
         /// <summary>
         /// Gets or sets a method to call the initialize the runtime platform services (e. g. AssetLoader)
         /// </summary>
-        public Action RuntimePlatformServices { get; set; }
+        public Action RuntimePlatformServicesInitializer { get; private set; }
 
         /// <summary>
         /// Gets or sets the <see cref="Application"/> instance being initialized.
         /// </summary>
-        public Application Instance { get; set; }
+        public Application Instance { get; protected set; }
 
         /// <summary>
         /// Gets or sets a method to call the initialize the windowing subsystem.
         /// </summary>
-        public Action WindowingSubsystem { get; set; }
+        public Action WindowingSubsystemInitializer { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the name of the windowing subsystem to use.
+        /// </summary>
+        public string WindowingSubsystemName { get; set; }
```

Comment:
In current implementation we have some code that looks for initializer by dll name - https://github.com/AvaloniaUI/Avalonia/blob/028c5c2f2fb156ad21d05642541e410662c4f1eb/src/Avalonia.Controls/AppBuilderBase.cs#L133
https://github.com/AvaloniaUI/Avalonia/blob/028c5c2f2fb156ad21d05642541e410662c4f1eb/src/Avalonia.Controls/AppBuilderBase.cs#L153

Having two different properties seems to be bug prone and counterintuitive.